pkgdb/src/versions.cc

```diff
@@ -232,19 +208,17 @@ cleanRange( const std::string & range )
 std::list<std::string>
 semverSat( const std::string & range, const std::list<std::string> & versions )
 {
-  std::list<std::string> args
-    = { "--include-prerelease", "--loose", "--range", cleanRange( range ) };
-  for ( const auto & version : versions ) { args.push_back( version ); }
-  auto [ec, lines] = runSemver( args );
-  /* TODO: determine parse error vs. empty list result. */
-  if ( ! nix::statusOk( ec ) ) { return {}; }
   std::list<std::string> rsl;
-  std::stringstream      oss( lines );
-  std::string            line;
-  while ( std::getline( oss, line, '\n' ) )
+  auto                   r = cleanRange( range );
+  for ( const auto & version : versions )
     {
-      if ( ! line.empty() ) { rsl.push_back( std::move( line ) ); }
+      if ( semver::valid( version ) && semver::satisfies( version, r ) )
+        {
+          rsl.push_back( version );
+        }
     }
+  rsl.sort( [&]( const auto & a, const auto & b )
+            { return semver::lt( a, b ); } );
```

Comment:
nonblocking: we sort in SQL which also handles preferPreReleases, so it's confusing to sort here
```suggestion
```